### ISSUE TITLE
Playlist columns resize consistently after every restart

### ISSUE THREAD

**Issue by kode54** [fooyin/fooyin#166]
**Describe the bug**
After every restart with basic layout with a playlist with headers, the "Playing" column resizes to fill the entire view, and all the other columns shrink to smaller and identical size portions to each other. I still need to test today's updates and see if they fix it.

**To Reproduce**
1. Install the player
2. Set the preset to basic layout with library panel and artwork on the side.
3. Add an album to the playlist
4. Quit
5. Restart

**Expected behaviour**
Playlist should maintain its last sizing state.

**Screenshots**
I'll post some screen shots and a copy of my config when I'm on the computer again.

**System:**
 - OS: Arch Linux
 - fooyin version: r2407.4449d2d0-1

**Additional context**
Just starting the preliminary report. I'll add note and edit more into it (and note it in replies) when I'm on my desktop again.

**Comment by kode54** [comment]
Okay, r2441.10fa7400-1 has changed the behavior. Now I restart to find that the "Playing" column has become the rightmost column of the playlist. None of the other columns remember their widths and always set themselves to a fixed width. Except for the rightmost column of the playlist, which always resizes itself to the entire width of the current playlist viewport.

Reordering the playlist columns, whatever becomes the rightmost column automatically resizes its width to the exact width of the playlist viewport, and cannot be resized.

Qt version 6.7.2-1.

**Comment by ludouzi** [comment]
Thanks for the report. I still can't reproduce this yet, but this gives me some clues. 

A good way to narrow this down will be to turn off auto-size sections in the header context menu. If the section widths are remembered, then the issue lies in the stretching behaviour. If not, it's a problem with the state.

**Comment by therealfumbles** [comment]
I just updated to commit 9a9acf0 and the playlist columns are messed up. The artist column is being moved to the end on reload.

Here is how it looks by default:
![Screenshot from 2024-07-04 15-17-17](https://github.com/fooyin/fooyin/assets/86084089/277bb632-5b5c-4963-ba80-f29922e3b2c2)

And after a relaunch:
![Screenshot from 2024-07-04 15-17-39](https://github.com/fooyin/fooyin/assets/86084089/5886e306-141a-4215-b055-6d373b0a0aa8)

If I turn off auto-size I get this after a relaunch:
![Screenshot from 2024-07-04 15-20-47](https://github.com/fooyin/fooyin/assets/86084089/282c051a-25dc-41de-8400-83a9cf2cabfa)

To get it back to how it should look I just reload the layout via View -> Layout -> my saved layout

**Comment by ludouzi** [comment]
I still can't reproduce this, but it's definitely got something to do with restoring the state. Would you be able to post the layout file for the above? Or just the header state value:

![image](https://github.com/fooyin/fooyin/assets/45490980/ddca7f21-4cd7-4da3-a63a-769ceee38225)

**Comment by therealfumbles** [comment]
Here's the layout:

[My Layout.fyl.txt](https://github.com/user-attachments/files/16100269/My.Layout.fyl.txt)

**Comment by therealfumbles** [comment]
With 9a9acf0 I get this:

Failed to load "fooyin" translations for locale "C" from "/usr/share/fooyin/translations"
"[PlaylistView] Restoring state for 1 section(s)"

The Failed to load translations line happens with 2f0ffee as well but the PlaylistView line is new.

**Comment by ludouzi** [comment]
The translations line can be ignored for now. 

The problem is that the playlist in your layout has 7 columns but it only sees (and restores) 1 column. This is likely due to the header not being notified in time of the section count on startup. This gives me some good ideas on a fix.

And yeah, artwork loading was fixed with https://github.com/fooyin/fooyin/commit/db30288a836c50db0fa3e2464732142a507f20cf, so they'll be some issues before then.

**Comment by therealfumbles** [comment]
It's better. The layout isn't shifting around anymore, but the first column (Playing) is being shrunk on launch. There's also a horizontal scroll bar at the bottom of the playlist as a result of the layout not being exactly right.

How it looks:
![Screenshot from 2024-07-04 16-56-47](https://github.com/fooyin/fooyin/assets/86084089/0d031a70-ba4e-4980-8962-cc0d90534837)

How it should look (fixed by reloading layout):
![Screenshot from 2024-07-04 16-56-59](https://github.com/fooyin/fooyin/assets/86084089/acb5bebc-acad-44cf-9692-51ce642772a3)

The terminal prints out: "[PlaylistView] Restoring state for 7 section(s)"

**Comment by ludouzi** [comment]
At least we're making progress. As it's fine when reloading the layout, it's got to be related to application startup. Does https://github.com/fooyin/fooyin/commit/547f41582a749848e50fb20b342d5dfaa23951a4 help?

**Comment by therealfumbles** [comment]
No, it's acting the same as above.

**Comment by therealfumbles** [comment]
Also, I don't know if this is relevant to this discussion or whether it should be in its own topic, but I save my layout with remember window size enabled. When I open fooyin it adds 2 extra pixels to the depth. With my layout it means I can see the tops of the next listing in the library tree.

When I reload the layout it resizes the window to the correct size.

**Comment by therealfumbles** [comment]
I just tried e74cc78 and the playlist is loading fine now. 

The first time I launched it it was still messed up so I uninstalled fooyin and then reinstalled from the new deb. I had just been installing it over the existing install and wanted to check if this might have been a problem.

**Comment by ludouzi** [comment]
> I just tried https://github.com/fooyin/fooyin/commit/e74cc78ca515f0af7c3a5428905dbd65e71bf56b and the playlist is loading fine now.

Good to hear! I'll leave this issue open until the original issue by @kode54 is resolved.

> The first time I launched it it was still messed up

In fooyin, layouts are only used to overwrite the current layout. On closing, it saves to a default layout file which is then restored on startup so that's probably why. 

This is also related to the window sizing issue you mentioned above. On startup, it uses `Settings->General->Startup->Behaviour` to determine how to show the main window. If it's not set to `Remember from last run`, it will show as a normal 1280x720 window.

**Comment by therealfumbles** [comment]
> In fooyin, layouts are only used to overwrite the current layout. On closing, it saves to a default layout file which is then restored on startup so that's probably why.

Yep, I thought it might have been because I was installing on top of the older version. But I think I just needed to reload the saved layout in order to update the default one. Then it loaded fine on subsequent launches.

> This is also related to the window sizing issue you mentioned above. On startup, it uses `Settings->General->Startup->Behaviour` to determine how to show the main window. If it's not set to `Remember from last run`, it will show as a normal 1280x720 window.

With Startup Behaviour set to Normal it was loading with my set size + 2 pixels, not 1280x720. But with it set to Remember it is now loading without those extra 2 pixels.

I diff-ed my saved layout with the default one and the only difference is the default one doesn't store the window size data. I tried copying my saved one over it but the window size data gets wiped once I run fooyin.

**Comment by kode54** [comment]
My issue is also resolved.
